### PR TITLE
Improve Hydra docs

### PR DESCRIPTION
**Pull request by kevgo**
I reorganized the OAuth2 and OpenID Connect introductions a bit. This topic is so complex that I think there should be two parts:

1. a very simple (idiot-proof) and short explanation of what OAuth is, to prove that we can understand it so well that we can summarize it properly for non-experts. 
2. a detailed discussion of the various parts of the OAuth spec and how Hydra satisfies and implements them, to prove that we understand all the details of it.

I think the intro in `/docs/hydra/index.md` is a good part for (1). Hence I reformat it in this PR. The structure is now:
- our summary of what OAuth 2.0 is
- our summary of what OpenID Connect is
- links to external videos and articles to learn more details if needed

What do you think?

### CODE REVIEW

File: docs/hydra/index.md [hunk 1]
```diff
@@ -16,160 +16,126 @@ are.
...
-
-Source: [IETF RFC 6749](https://tools.ietf.org/html/rfc6749)
+specified in [IETF RFC 6749](https://tools.ietf.org/html/rfc6749). OAuth 2.0
+enables a third-party application to obtain limited access to resources on an
+HTTP server on behalf of the owner of those resources.
+
+Why is this important? Without OAuth 2.0, a resource owner who wants to share
+resources in their account with a third party would have to share their
+credentials with that third party. As an example, let's say you (a resource
+owner) have some photos (resources) stored on a social network (the resource
```
Comment (kevgo) [inline docs/hydra/index.md]: This example is super helpful! I changed the "photo sharing service" in it to "social network". 

- I found that "photo sharing service" and "printing service" sound too alike. I kept confusing the two. In the real world, these two functionalities (sharing and printing photos) often go together. On the other hand, it seems more plausible that a social network has lots of photos but not necessarily a way to print them directly.
- We want to provide an example for other information (besides photos) that isn't included in the OAuth scopes. That's hard for a service that only shares photos.

File: docs/hydra/index.md [hunk 2]
```diff
@@ -16,160 +16,126 @@ are.
...
+server). Now you want to print them using a third-party printing service. Before
+OAuth 2.0 existed, you would have to enter your social network password into the
+printing service so that it can access and print your photos. Sharing secret
+passwords with third parties is obviously very problematic.
+
+OAuth addresses this problem by introducing:
+
+- the distinction between resource ownership and resource access for clients
+- the ability to define fine-grained access privileges (called OAuth scopes)
```
Comment (kevgo) [inline docs/hydra/index.md]: I added this point to the docs

File: docs/hydra/index.md [hunk 3]
```diff
@@ -16,160 +16,126 @@ are.
...
+  instead of full account access for third parties
+- an authorization layer and workflow that allows resource owners to grant
+  particular clients particular types of access to particular resources.
+
+With OAuth, clients can request access to resources on a server, and the owner
+of these resources can grant the requested access together with dedicated
+credentials. In our example, you could grant the printing service read-only
+access to your photos (only your photos, not your friend list) on the social
```
Comment (kevgo) [inline docs/hydra/index.md]: OAuth scope is only for photos, not for the friend list

File: docs/hydra/index.md [hunk 4]
```diff
@@ -16,160 +16,126 @@ are.
...
+
+OpenID Connect allows clients of all types, including web and mobile, to receive
+information about authenticated sessions and end users. The specification is
+extensible, allowing participants to add encryption of identity data, discovery
+of OpenID Providers, and session management as needed.
+
+There are different work flows for OpenID Connect 1.0. We recommend checking out
 the OpenID Connect sandbox at [openidconnect.net](https://openidconnect.net/).
 
+A more detailed introduction of both OAuth 2.0 and OpenID Connect is available
```
Comment (kevgo) [inline docs/hydra/index.md]: The links to external videos are now down here, below the (shortened) summaries of OAuth2 and OpenID Connect.

File: docs/hydra/index.md (hunk 1 above)
Comment (aeneasr) [reply docs/hydra/index.md]: Awesome!

**Comment by aeneasr** [comment]
Awesome Kevin, thank you!

**Pull request merged**